### DISCUSSION TITLE
Blog

### DISCUSSION THREAD

**Discussion by Yahya-EzEd** [AlinaBoess/SoftwareEngineeringProjektTINF23B5#1] [General]

**Comment by Yahya-EzEd** [comment]
# Erster Blogeintrag

Herzlich Willkommen zu unserem Software Engineering Projekt.

## Teamvorstellung
Wir sind ein Team aus 5 Entwicklern:
- Moumem
- Yahya
- Lukas
- Alex
- Alina

Unsere Rollen haben wir noch nicht konkret aufgeteilt, da wir noch am Anfang der Entwicklung stehen und noch sehr flexibel sind.

## Projektvision
Wir möchten für unser Projekt ein Restaurantreservierungssystem erstellen. Hierfür entwickeln wir eine Website. Hier soll es möglich sein, dass sich ein User ein Restaurant aussuchen kann und dort dann einen Raumplan mit den verschiedenen Tischen sieht. Er kann dort dann freie Tische wählen und sie sich reservieren lassen. Er bekommt dann eine Bestätigung über seine Reservierung. Es soll auch möglich sein diese Reservierung wieder zu stonieren. Im Anschluss soll es auch möglich sein Feedback zu geben.

## Technologie und Plattformen 
Wir haben uns für C# als Programmiersprache entschieden in Kombination mit ASP.NET Core und Visual Studio als IDE. Damit haben wir eine gute Basis für die Entwicklung.
Für das Frontend nehmen wir HTML, CSS und JavaScript/TypeScript.
Die Installation von Visual Studio, die Einrichtung von GitHub und weiteren Tools haben wir bereits abgeschlossen.

## Scrum Setup
Für unser Software Engineering Projekt haben wir uns für Scrum entschieden, um unsere Arbeit gut zu strukturieren und regelmäßig Ergebnisse liefern zu können. Die Sprintlänge beträgt zwei Wochen.

Unsere Scrum-Termine sehen folgendermaßen aus:
- Sprintwechsel-Termine (Review, Retrospektive, Planning): Diese finden am Ende jedes Sprints statt und dauern jeweils etwa eine Stunde. Dabei besprechen wir, was gut gelaufen ist, wo es Verbesserungsbedarf gibt und planen den nächsten Sprint.
- Dailys: Um auf dem Laufenden zu bleiben, haben wir zweimal pro Woche (montags und donnerstags) Dailys, die etwa 15 bis 30 Minuten dauern. Hier gibt jeder einen kurzen Überblick darüber, was erledigt wurde, was ansteht und ob es irgendwelche Probleme gibt.


### Einrichtung unserer Scrum-Plattform

Wir nutzen Jira als unsere Scrum-Plattform, und die wichtigsten Dinge sind schon fast komplett eingerichtet:

- Boards und Sprint-Logs: Unsere Aufgaben sind in Jira übersichtlich organisiert, sodass jeder sehen kann, was ansteht und wie weit wir sind.

- Backlog: Der Backlog ist soweit fertig und enthält alle Stories und Aufgaben, die wir in den nächsten Sprints umsetzen wollen.

- Zeitaufwands-Tracking: Es gibt eine Seite in Jira, auf der wir den Zeitaufwand für jede Aufgabe erfassen können. Das hilft uns, die Arbeitszeiten im Blick zu behalten.

Unser Workflow ist aktuell noch in Arbeit. Wir werden ihn in den nächsten Wochen weiter anpassen, um die Prozesse klarer zu definieren.


## Sprint Backlog für Sprint 1
**Unser Backlog: Erste Aufgaben und Planung**
Für den ersten Sprint haben wir uns auf grundlegende Aufgaben konzentriert:

- **Projektidee und Funktionen ausarbeiten:** Wir haben definiert, was unser Restaurant-Reservierungstool können soll, und die wichtigsten Features festgelegt.

- **Technologie-Stack auswählen:** Wir haben uns für C# als Programmiersprache entschieden und Visual Studio als IDE. Damit haben wir eine gute Basis für die Entwicklung.

- **Tool-Setup:** Die Installation von Visual Studio, die Einrichtung von GitHub und weiteren Tools haben wir bereits abgeschlossen.

- **Einarbeitung in C#: Da einige Teammitglieder noch nicht viel Erfahrung mit C# haben, gibt es eine Story für die Einarbeitung. Auch der Umgang mit Git wurde als separate Aufgabe definiert, um die Versionskontrolle sicherzustellen.**

- **Datenbankdesign:** Ein weiteres Ziel ist es, erste Entwürfe für das Datenbankdesign zu erstellen.

Die Aufwände für die meisten Stories im ersten Sprint wurden bereits geschätzt. Allerdings ist die Definition von "Done" bei einigen Themen noch nicht eindeutig. Sobald alle die nötigen Kenntnisse erworben haben, können die Stories als abgeschlossen betrachtet werden.

**Erstes Sprint-Meeting: Rollen, Aufgaben und Design**

In unserem ersten Sprint-Meeting haben wir die Projektidee konkretisiert und die Stories für den ersten Sprint erstellt. Die meisten Aufgaben wurden vorerst allgemein verteilt, da sie von allen bearbeitet werden müssen. Spezifischere Aufgaben folgen dann in den nächsten Sprints.

Was wir sonst noch besprochen haben:

- Rollenverteilung im Team, damit klar ist, wer welche Verantwortung hat.

- Erste Ideen fürs Design des Tools.

- Planung der Datenbankstruktur, um eine solide Basis für die weitere Entwicklung zu schaffen.


Hier ist der aktuelle Zustand unseres Boards: 
<img width="746" alt="Screenshot 2024-10-14 174123" src="https://github.com/user-attachments/assets/4b8a9734-0515-4dcc-981c-4e096d9eb683">

**Comment by SchloesserJonas** [comment]
Hi,

euer Scrum Setup klingt auf jeden Fall schon mal sehr durchdacht! 😄

Im Bezug auf eure Projektidee wäre es interessant zu erfahren welche Kernfunktionalitäten ihr für eure Idee identifiziert habt.

Was waren die Ergebnisse eures ersten Sprint Meetings?
Wer übernimmt welche Rolle im Projekt?
Was für ein Konzept verfolgt ihr beim Design?

Wir sind gespannt wie es bei euch weiter geht!

GaLiGrü 😊
Paula, Safae, Samuel, Valentin, Jonas

> Green Sprout 🌱

**Reply by AlinaBoess** [reply]
Hi,

Vielen Dank für euer erstes Feedback! Wir sind gerade noch dabei den Blogeintrag etwas zu überarbeiten, schaut gerne später nochmal vorbei.

Liebe Grüße
Moumen, Yahya, Alex, Lukas, Alina

**Comment by GermanJesus-lul** [comment]
Hey,
Info: Euren Post habt ihr als Kommentar an eure Diskussion veröffentlicht. Über den Drei-Punkte-Button (Popup: "Discussion options") könnt ihr "Edit" auswählen und den Diskussionspost nochmal abändern.

Finde die Idee mit dem Raumplan schonmal super. Wird es auch eine entsprechenden Anwendungsteil für Restaurantbesitzer geben?

Mit freundlichen Grüßen,
Julius, Luca, Jaron, Tom
